### ISSUE TITLE
Uncommitted generated code makes testing code unbuildable without go generate

### ISSUE THREAD

**Issue by zikaeroh** [maxbrunsfeld/counterfeiter#126]
All of the testing for this repo assumes that `go generate ./...` gets run first. Unfortunately, this means that there is at least one package imported from a test that does not exist (`fixturesfakes`) for a "normal" checkout of the repo.

If you use counterfeiter in GOPATH mode, this isn't a big deal; `go get` won't tell you that the tests are not buildable.

However, when used from a module (for example, when using the recommended `tools.go` pattern with `gobin`), the Go tooling tries to search for packages/modules that fit the bad import, leading to a lot of printed text, hidden errors, and retried work whenever `go.mod` needs to be modified. See golang/go#32720.

```
$ go mod tidy     
go: finding github.com/maxbrunsfeld/counterfeiter/v6/fixtures/fixturesfakes latest
go: finding github.com/maxbrunsfeld/counterfeiter/v6/fixtures latest
go: finding github.com/maxbrunsfeld/counterfeiter latest
go: downloading github.com/maxbrunsfeld/counterfeiter v0.0.0-20190614042641-5e26f8d4cfe9
```

```
$ go list -tags=tools all
go: finding github.com/maxbrunsfeld/counterfeiter/v6/fixtures/fixturesfakes latest
go: finding github.com/maxbrunsfeld/counterfeiter/v6/fixtures latest
go: finding github.com/maxbrunsfeld/counterfeiter latest
go: downloading github.com/maxbrunsfeld/counterfeiter v0.0.0-20190614042641-5e26f8d4cfe9
can't load package: package github.com/maxbrunsfeld/counterfeiter/v6/fixtures/fixturesfakes: unknown import path "github.com/maxbrunsfeld/counterfeiter/v6/fixtures/fixturesfakes": cannot find module providing package github.com/maxbrunsfeld/counterfeiter/v6/fixtures/fixturesfakes
```

Can this code be committed such that the project is fully buildable by default?

Alternatively, if this generation is being used to test `counterfeiter`'s generated code itself (since I know this is a code generator) and the code cannot be committed, it might be good to consider reworking the tests to do things out of testdata or similar, such that one can pull the repo and just run `go test ./...`, preventing the Go tooling from complaining. (For example, using [txtar/testscripts](https://godoc.org/github.com/rogpeppe/go-internal/testscript), or similar.)

**Comment by joefitzgerald** [comment]
@zikaeroh yes this makes sense. We can add a step in the build to validate that `go generate ./...` results in no changes to the working copy.

**Comment by zikaeroh** [comment]
I think the first step is going to be to remove what's needed for the build from the gitignore. I've found that simply removing the line containing `fixtures/fixturesfakes/` and then doing:

```
$ go generate ./...
$ git add .
$ git clean -fdx
```

Leaves the repo in a buildable state in a staged commit, so that's a start.

**Comment by zikaeroh** [comment]
Tried my hand at a fix in #127.
